scanpipe/pipelines/populate_purldb.py

```diff
@@ -48,6 +50,18 @@ def populate_purldb_with_discovered_dependencies(self):
             package_type="DiscoveredDependency",
         )
 
+    def populate_purldb_with_detected_purls(self):
```

Comment:
It's unclear why this is only called when the project has not packages/dependencies.

Reply:
This step exists to get purls out of package_data saved at resources. These are still detected and saved, even though we don't have Package/Dependency instances (the assemble=False skips the package assembly, but resource level package detection is still enabled and happening).

Reply:
@tdruez I've added this in the code as comments, thanks!